src/server/helpers/middleware.js

```diff
@@ -211,3 +211,57 @@ export function makeCollectionLoader() {
 		}
 	};
 }
+
+export async function validateBBIDsForCollectionAdd(req, res, next) {
+	const {Entity} = req.app.locals.orm;
+	const {bbids = []} = req.body;
+	if (!bbids.length) {
+		return next(new error.BadRequestError('BBIDs array is empty'));
+	}
+	const {collection} = res.locals;
+	const collectionType = collection.entityType;
+	for (let i = 0; i < bbids.length; i++) {
+		const bbid = bbids[i];
+		if (!commonUtils.isValidBBID(bbid)) {
+			return next(new error.BadRequestError(`Invalid BBID ${bbid}`, req));
+		}
+	}
+	const entities = await new Entity().where('bbid', 'in', bbids).fetchAll({require: false});
+	const entitiesJSON = entities.toJSON();
+	for (let i = 0; i < bbids.length; i++) {
+		const bbid = bbids[i];
+		const entity = entitiesJSON.find(currEntity => currEntity.bbid === bbid);
+		if (!entity) {
+			return next(new error.BadRequestError(`${collectionType} ${bbid} does not exist`, req));
```

Comment:
I hadn't thought about this before, but I guess we should send an `error.NotFoundError`  (404) here instead.